src/Controller/CRUDController.php

```diff
@@ -368,15 +368,15 @@ public function batchAction(Request $request)
         if ($data = json_decode($request->get('data', ''), true)) {
             $action = $data['action'];
             $idx = $data['idx'];
-            $allElements = $data['all_elements'];
+            $allElements = (bool) $data['all_elements'];
             $request->request->replace(array_merge($request->request->all(), $data));
         } else {
             $request->request->set('idx', $request->get('idx', []));
             $request->request->set('all_elements', $request->get('all_elements', false));
 
             $action = $request->get('action');
             $idx = $request->get('idx');
-            $allElements = $request->get('all_elements');
+            $allElements = (bool) $request->get('all_elements');
```

Comment:
What about using `$request->request->getBoolean('all_elements')`? There is other get just above that could be changed too.

This code is a bit strange BTW, it first get the all_elements, then sets it to the request with the same value, and then get it again to assign it to a var. 🤔